### DISCUSSION TITLE
Automatic and periodic checking for upgrades

### DISCUSSION THREAD

**Discussion by sponz** [arendst/Tasmota#13321] [Q&A] [answered]
Is there any way to check automatically and periodically for upgrades on the server? 

The idea would be that the device checks every week if there is a new version and upgrades itself automatically. 

Thanks!

**Comment by ascillato** [comment, marked as answer]
There is no such feature in Tasmota. Also, it is not a good idea to automatically update your device without YOU checking the breaking changes.
If a new feature or fix is useful for you, it is good to update. If not, it is better not to update and leave your working setup as it is.

**Comment by sponz** [comment]
Thanks for your reply. I am thinking about having my own upgrade server, so I can manage it "locally" and I don't have to worry about sending upgrade mqtt messages to force a manual upgrade.

**Reply by sfromis** [reply]
As breaking changes do occur (not very often, but not out of the question), doing automatic updates might leave with you with devices not working as intended.

**Reply by barbudor** [reply]
I also have my own server as I am using binaries with my key settings already hardcoded (wifi & mqtt credentials)
But I always trigger the upgrades manually
Starting with 1 non critical device, and then spreading around the house on a 1 week period. I have some ready batch files to send `upgrade 1` commands to groups of devices
I always follow production release even if there is no immediate gain vs "don't touch something that works" because:
1) I'm geeky so I want the very last ;)
2) I want to have an homogenous revision on all devices to avoid caring for different behavior (like changes in JSON format of some commands as this occured in the past months)
3) big step upgrades may be painful and more risky than progressive

**Reply by sfromis** [reply]
I'm also dancing on the bleeding edge with updates, but it nearly always works out, even without being as slow as a whole week. Of course, I'm very aware of the potential for needing to change my tools to have everything working.

Unless you want this show, I'd really suggest you to stay with what works, as improvements with new releases are almost always something you need to actively use, meaning that you'd not get that gain from "blindly" updating to a newer firmware version.
